addons/account/wizard/account_payment_register.py

```diff
@@ -834,7 +863,18 @@ def _reconcile_payments(self, to_process, edit_mode=False):
 
     def _create_payments(self):
         self.ensure_one()
-        batches = self._get_batches()
+        all_batches = self._get_batches()
+        batches = []
+        # Skip batches that are not valid (bank account not trusted but required)
+        for batch in all_batches:
+            batch_account = self._get_batch_available_partner_banks(batch, self.journal_id)
+            if self.require_partner_bank_account and not batch_account.allow_out_payment:
+                continue
+            batches.append(batch)
+
+        if not batches:
+            raise UserError(_('To record %s payments, the recipient bank account must be trusted. Go to the bank account to validate them first.', self.payment_method_line_id.name))
```

Comment:
```suggestion
            raise UserError(_('To record payments from %s, the recipient bank account must be manually validated. Go to the bank account to specify them first.', self.payment_method_line_id.name))
```
Not clear what the "go to the bank account" mean. Should be rephrased.